go/pkg/client/cas.go

```diff
@@ -91,10 +93,40 @@ func (c *Client) uploadProcessor() {
 				}
 				return
 			}
-			buffer = append(buffer, req)
-			if len(buffer) >= casChanBufferSize {
-				c.upload(buffer)
-				buffer = nil
+			if !req.cancel {
+				buffer = append(buffer, req)
+				if len(buffer) >= casChanBufferSize {
+					c.upload(buffer)
+					buffer = nil
+				}
+				continue
+			}
+			// Cancellation request.
+			var newBuffer []*uploadRequest
+			for _, r := range buffer {
+				if r.c != req.c || r.wait != req.wait {
+					newBuffer = append(newBuffer, r)
+				}
+			}
+			buffer = newBuffer
+			st, ok := c.casUploads[req.c.Digest()]
+			if ok {
+				st.mu.Lock()
+				var newClients []chan<- *uploadResponse
```

Comment:
Can you add a comment clarifying newClients? Something along the lines of it being a list of the remaining clients after the current cancellation requestor drops out?

Reply:
Did one better and renamed the variable so it's clear.